pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -262,6 +259,17 @@ func (pas *PodAutoscalerStatus) inStatusFor(status corev1.ConditionStatus, now t
 	return now.Sub(cond.LastTransitionTime.Inner.Add(dur))
 }
 
+// inStatusHasBeenActiveFor returns positive duration if the PodAutoscalerStatus's HasBeenActive condition
+// has stayed in the specified status for at least the specified duration. Otherwise it returns negative
+// duration, including when the status is undetermined (Active condition is not found.)
+func (pas *PodAutoscalerStatus) inStatusHasBeenActiveFor(status corev1.ConditionStatus, now time.Time, dur time.Duration) time.Duration {
+	cond := pas.GetCondition(PodAutoscalerConditionHasBeenActive)
+	if cond == nil || cond.Status != status {
+		return -1
+	}
+	return now.Sub(cond.LastTransitionTime.Inner.Add(dur))
+}
+
```

Comment:
At times I think this should part of the condition itself
cc @mattmoor @whaught